### ISSUE TITLE
Choose hyperparameters better for each test

### ISSUE THREAD

**Issue by ijkilchenko** [ijkilchenko/Ann#2]
Since the cost surface is not convex, random initialization can actually lead gradient descent to not convert sometimes. Some tests (and maybe the demo) should be adjusted by choosing hyperparameters better and maybe doing several random initializations...
